src/frontend/src/lib/canisters/signer.canister.ts

```diff
@@ -0,0 +1,29 @@
+import type { BitcoinNetwork, _SERVICE as SignerService } from '$declarations/signer/signer.did';
+import { idlFactory as idlCertifiedFactorySigner } from '$declarations/signer/signer.factory.certified.did';
+import { idlFactory as idlFactorySigner } from '$declarations/signer/signer.factory.did';
+import { Principal } from '@dfinity/principal';
+import { Canister, createServices, type CanisterOptions } from '@dfinity/utils';
+
+interface SignerCanisterOptions<T> extends Omit<CanisterOptions<T>, 'canisterId'> {
+	canisterId: Principal;
+}
+
+export class SignerCanister extends Canister<SignerService> {
```

Comment:
Is there a particular reason why you developed this with a class and specific pattern? Are you planing to re-implement  the module`actors.ic` later on?

If not, we can also resolve it with a function and expose `createActor`.

```
export const getSignerWorkerActor = async ({
	identity,
	signerCanisterId,
	certified = true
}: {
	identity: OptionIdentity;
	certified?: boolean;
	signerCanisterId: CanisterIdText;
}): Promise<SignerActor> => {
	assertNonNullish(identity);

	return await createActor<SignerActor>({
		canisterId: signerCanisterId,
		idlFactory: certified ? idlCertifiedFactorySigner : idlFactorySigner,
		identity
	});
};
```
 
But not against the class pattern, we should just be consistent and rework existing actors, might be a good idea.

Reply:
No particular reasons, I was mostly inspired by how canisters implemented in `ic-js`. In general, I think it would be nice if we follow the same "class" pattern for actors in oisy. It doesn't involve a lot of refactorings as, from what I see, we'll only need to rework the backend actor as well as to extend the one that I implemented for signer with other methods that are already in use. WDYT?

Reply:
That’s generally not the approach we've taken in the NNS dapp or Oisy; we used this pattern in ic-js for historical reasons.

That said, I'm not opposed to it, and I can see the value if we want to expand further by introducing scoped caching.

If you’d like to go down this route, we can start by adapting the existing code and transforming `getSignerActor` into a Class-based approach in a separate PR, sSince the current PR adds new code without modifying existing functionality and also since we discussed scope in comments. Note that the pattern should account for multiple actors (for UI or worker contexts) and maintain the caching features. It may also be worthwhile to add tests if we decide to proceed too I would say.

Reply:
Sounds like a plan! Shall we then merge this branch so I can use the SignerCanister class for the further refactorings we discussed + for the next BTC integration PRs?

Reply:
>  Shall we then merge this branch

I rather like not. I think we should not introduce the new pattern this way but, cleanly implement it.

Reply:
I created a PR based on this one where I used the canister approach for all the existing signer API functions - https://github.com/dfinity/oisy-wallet/pull/2425.

Reply:
The code from this PR is now a part of https://github.com/dfinity/oisy-wallet/pull/2425. We can continue the discussions there.